### PR TITLE
Feature/update database

### PR DESCRIPTION
**Pull request by emarkees**
## In this milestone
- I implement the queries and the transaction.
- I added screenshots with the results of the queries and play transactions.

Kindly review and give me a star ⭐ ⭐ 

Screenshot (396)
Screenshot (397)
Screenshot (398)
Screenshot (399)
Screenshot (400)
Screenshot (401)
Screenshot (390)
Screenshot (391)
Screenshot (392)
Screenshot (393)
Screenshot (394)
Screenshot (395)
Screenshot (389)
Screenshot (388)

Kindly review and give me a star ⭐ ⭐

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -7,4 +7,94 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
...
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+/* Inside a transaction update the animals table by setting the species column to unspecified */
+BEGIN;
+
+UPDATE animals SET species = 'unspecified';
+
+ROLLBACK;
+
```
Comment (laiifuu) [inline queries.sql]: As requested in the [requirements](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=Verify%20that%20change%20was%20made.%20Then%20roll%20back%20the%20change%20and%20verify%20that%20the%20species%20columns%20went%20back%20to%20the%20state%20before%20the%20transaction.) you should verify that the change was made, you can do so with a `SELECT` query. You should also verify that the species columns went back to its initial state after the rollback, you can do so with another `SELECT` query.

File: queries.sql [hunk 2]
```diff
@@ -7,4 +7,94 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
...
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+
+/* To verify changes */
+
+SELECT * FROM animals;
+
+ROLLBACK;
+
+COMMIT;
+
```
Comment (laiifuu) [inline queries.sql]: For this transaction, you are not supposed to `ROLLBACK` before commiting. Also, you should verify that the changes were made after the `COMMIT`.

File: queries.sql [hunk 3]
```diff
@@ -7,4 +7,94 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
...
+/* To delete all records in a transaction */
+
+BEGIN;
+
+DELETE FROM animals;
+
+ROLLBACK;
+
+COMMIT;
```
Comment (laiifuu) [inline queries.sql]: For this transaction, it is not necessary to `COMMIT` the changes.

File: queries.sql [hunk 4]
```diff
@@ -7,4 +7,94 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
...
+
+ROLLBACK TO SAVEPOINT del_dob_animals;
+
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+
+COMMIT;
+
+/* To verify changes */
+
+SELECT * FROM animals;
```
Comment (laiifuu) [inline queries.sql]: There's no need to check for changes at this point.

**Review by laiifuu — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED :recycle: 

## Hi @emarkees :wave: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
- [x] The SQL commands were added to the proper files :heavy_check_mark: 
- [x] Organized code :+1: 

## Required Changes ♻️
_Check the comments under the review :point_down:_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 4 above)
Comment (emarkees) [reply queries.sql]: done

File: queries.sql (hunk 3 above)
Comment (emarkees) [reply queries.sql]: done

File: queries.sql (hunk 2 above)
Comment (emarkees) [reply queries.sql]: done

File: queries.sql (hunk 1 above)
Comment (emarkees) [reply queries.sql]: done

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🍾✔🎉**
Hello @emarkees 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- Descriptive PR.
- Good commit messages.
- Correct git-flow.

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by emarkees** [comment]
Thank you @emmanuellekamwa

**Pull request merged**